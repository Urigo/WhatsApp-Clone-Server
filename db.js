"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
// For TypeORM
require("reflect-metadata");
const Chat_1 = require("./entity/Chat");
const Recipient_1 = require("./entity/Recipient");
const moment_1 = __importDefault(require("moment"));
const Message_1 = require("./entity/Message");
const User_1 = require("./entity/User");
var MessageType;
(function (MessageType) {
    MessageType[MessageType["PICTURE"] = 0] = "PICTURE";
    MessageType[MessageType["TEXT"] = 1] = "TEXT";
    MessageType[MessageType["LOCATION"] = 2] = "LOCATION";
})(MessageType = exports.MessageType || (exports.MessageType = {}));
async function addSampleData(connection) {
    const user1 = new User_1.User({
        username: 'ethan',
        password: '$2a$08$NO9tkFLCoSqX1c5wk3s7z.JfxaVMKA.m7zUDdDwEquo4rvzimQeJm',
        name: 'Ethan Gonzalez',
        picture: 'https://randomuser.me/api/portraits/thumb/men/1.jpg',
        phone: '+391234567890',
    });
    await connection.manager.save(user1);
    const user2 = new User_1.User({
        username: 'bryan',
        password: '$2a$08$xE4FuCi/ifxjL2S8CzKAmuKLwv18ktksSN.F3XYEnpmcKtpbpeZgO',
        name: 'Bryan Wallace',
        picture: 'https://randomuser.me/api/portraits/thumb/men/2.jpg',
        phone: '+391234567891',
    });
    await connection.manager.save(user2);
    const user3 = new User_1.User({
        username: 'avery',
        password: '$2a$08$UHgH7J8G6z1mGQn2qx2kdeWv0jvgHItyAsL9hpEUI3KJmhVW5Q1d.',
        name: 'Avery Stewart',
        picture: 'https://randomuser.me/api/portraits/thumb/women/1.jpg',
        phone: '+391234567892',
    });
    await connection.manager.save(user3);
    const user4 = new User_1.User({
        username: 'katie',
        password: '$2a$08$wR1k5Q3T9FC7fUgB7Gdb9Os/GV7dGBBf4PLlWT7HERMFhmFDt47xi',
        name: 'Katie Peterson',
        picture: 'https://randomuser.me/api/portraits/thumb/women/2.jpg',
        phone: '+391234567893',
    });
    await connection.manager.save(user4);
    const user5 = new User_1.User({
        username: 'ray',
        password: '$2a$08$6.mbXqsDX82ZZ7q5d8Osb..JrGSsNp4R3IKj7mxgF6YGT0OmMw242',
        name: 'Ray Edwards',
        picture: 'https://randomuser.me/api/portraits/thumb/men/3.jpg',
        phone: '+391234567894',
    });
    await connection.manager.save(user5);
    const user6 = new User_1.User({
        username: 'niko',
        password: '$2a$08$fL5lZR.Rwf9FWWe8XwwlceiPBBim8n9aFtaem.INQhiKT4.Ux3Uq.',
        name: 'Niccol√≤ Belli',
        picture: 'https://randomuser.me/api/portraits/thumb/men/4.jpg',
        phone: '+391234567895',
    });
    await connection.manager.save(user6);
    const user7 = new User_1.User({
        username: 'mario',
        password: '$2a$08$nDHDmWcVxDnH5DDT3HMMC.psqcnu6wBiOgkmJUy9IH..qxa3R6YrO',
        name: 'Mario Rossi',
        picture: 'https://randomuser.me/api/portraits/thumb/men/5.jpg',
        phone: '+391234567896',
    });
    await connection.manager.save(user7);
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(10, 'weeks').toDate(),
        allTimeMembers: [user1, user3],
        listingMembers: [user1, user3],
        messages: [
            new Message_1.Message({
                sender: user1,
                content: 'You on your way?',
                createdAt: moment_1.default().subtract(1, 'hours').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user3],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user3,
                    }),
                ],
            }),
            new Message_1.Message({
                sender: user3,
                content: 'Yep!',
                createdAt: moment_1.default().subtract(1, 'hours').add(5, 'minutes').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user3],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user1,
                    }),
                ],
            }),
        ],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(9, 'weeks').toDate(),
        allTimeMembers: [user1, user4],
        listingMembers: [user1, user4],
        messages: [
            new Message_1.Message({
                sender: user1,
                content: 'Hey, it\'s me',
                createdAt: moment_1.default().subtract(2, 'hours').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user4],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user4,
                    }),
                ],
            }),
        ],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(8, 'weeks').toDate(),
        allTimeMembers: [user1, user5],
        listingMembers: [user1, user5],
        messages: [
            new Message_1.Message({
                sender: user1,
                content: 'I should buy a boat',
                createdAt: moment_1.default().subtract(1, 'days').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user5],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user5,
                    }),
                ],
            }),
            new Message_1.Message({
                sender: user1,
                content: 'You still there?',
                createdAt: moment_1.default().subtract(1, 'days').add(16, 'hours').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user5],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user5,
                    }),
                ],
            }),
        ],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(7, 'weeks').toDate(),
        allTimeMembers: [user3, user4],
        listingMembers: [user3, user4],
        messages: [
            new Message_1.Message({
                sender: user3,
                content: 'Look at my mukluks!',
                createdAt: moment_1.default().subtract(4, 'days').toDate(),
                type: MessageType.TEXT,
                holders: [user3, user4],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user4,
                    }),
                ],
            }),
        ],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(6, 'weeks').toDate(),
        allTimeMembers: [user2, user5],
        listingMembers: [user2, user5],
        messages: [
            new Message_1.Message({
                sender: user2,
                content: 'This is wicked good ice cream.',
                createdAt: moment_1.default().subtract(2, 'weeks').toDate(),
                type: MessageType.TEXT,
                holders: [user2, user5],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user5,
                    }),
                ],
            }),
            new Message_1.Message({
                sender: user5,
                content: 'Love it!',
                createdAt: moment_1.default().subtract(2, 'weeks').add(10, 'minutes').toDate(),
                type: MessageType.TEXT,
                holders: [user2, user5],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user2,
                    }),
                ],
            }),
        ],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(5, 'weeks').toDate(),
        allTimeMembers: [user1, user6],
        listingMembers: [user1],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(4, 'weeks').toDate(),
        allTimeMembers: [user2, user1],
        listingMembers: [user2],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(3, 'weeks').toDate(),
        name: 'Ethan\'s group',
        picture: 'https://randomuser.me/api/portraits/thumb/lego/1.jpg',
        allTimeMembers: [user1, user3, user4, user6],
        listingMembers: [user1, user3, user4, user6],
        actualGroupMembers: [user1, user4, user6],
        admins: [user1, user6],
        owner: user1,
        messages: [
            new Message_1.Message({
                sender: user1,
                content: 'I made a group',
                createdAt: moment_1.default().subtract(2, 'weeks').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user3, user4, user6],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user3,
                    }),
                    new Recipient_1.Recipient({
                        user: user4,
                    }),
                    new Recipient_1.Recipient({
                        user: user6,
                    }),
                ],
            }),
            new Message_1.Message({
                sender: user1,
                content: 'Ops, Avery was not supposed to be here',
                createdAt: moment_1.default().subtract(2, 'weeks').add(2, 'minutes').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user4, user6],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user4,
                    }),
                    new Recipient_1.Recipient({
                        user: user6,
                    }),
                ],
            }),
            new Message_1.Message({
                sender: user4,
                content: 'Awesome!',
                createdAt: moment_1.default().subtract(2, 'weeks').add(10, 'minutes').toDate(),
                type: MessageType.TEXT,
                holders: [user1, user4, user6],
                recipients: [
                    new Recipient_1.Recipient({
                        user: user1,
                    }),
                    new Recipient_1.Recipient({
                        user: user6,
                    }),
                ],
            }),
        ],
    }));
    await connection.manager.save(new Chat_1.Chat({
        createdAt: moment_1.default().subtract(2, 'weeks').toDate(),
        name: 'Ray\'s group',
        allTimeMembers: [user3, user6],
        listingMembers: [user3, user6],
        actualGroupMembers: [user3, user6],
        admins: [user6],
        owner: user6,
    }));
}
exports.addSampleData = addSampleData;
