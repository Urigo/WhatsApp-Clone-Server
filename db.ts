export type User = {
  id: string;
  name: string;
  username: string;
  password: string;
  picture: string;
};

export type Message = {
  id: string;
  content: string;
  createdAt: Date;
  sender: string;
  recipient: string;
};

export type Chat = {
  id: string;
  messages: string[];
  participants: string[];
};

export const users: User[] = [];
export const messages: Message[] = [];
export const chats: Chat[] = [];

export const resetDb = () => {
  users.splice(
    0,
    Infinity,
    ...[
      {
        id: '1',
        name: 'Ray Edwards',
        username: 'ray',
        password:
          '$2a$08$NO9tkFLCoSqX1c5wk3s7z.JfxaVMKA.m7zUDdDwEquo4rvzimQeJm', // 111
        picture: 'https://randomuser.me/api/portraits/thumb/lego/1.jpg',
      },
      {
        id: '2',
        name: 'Ethan Gonzalez',
        username: 'ethan',
        password:
          '$2a$08$xE4FuCi/ifxjL2S8CzKAmuKLwv18ktksSN.F3XYEnpmcKtpbpeZgO', // 222
        picture: 'https://randomuser.me/api/portraits/thumb/men/1.jpg',
      },
      {
        id: '3',
        name: 'Bryan Wallace',
        username: 'bryan',
        password:
          '$2a$08$UHgH7J8G6z1mGQn2qx2kdeWv0jvgHItyAsL9hpEUI3KJmhVW5Q1d.', // 333
        picture: 'https://randomuser.me/api/portraits/thumb/men/2.jpg',
      },
      {
        id: '4',
        name: 'Avery Stewart',
        username: 'avery',
        password:
          '$2a$08$wR1k5Q3T9FC7fUgB7Gdb9Os/GV7dGBBf4PLlWT7HERMFhmFDt47xi', // 444
        picture: 'https://randomuser.me/api/portraits/thumb/women/1.jpg',
      },
      {
        id: '5',
        name: 'Katie Peterson',
        username: 'katie',
        password:
          '$2a$08$6.mbXqsDX82ZZ7q5d8Osb..JrGSsNp4R3IKj7mxgF6YGT0OmMw242', // 555
        picture: 'https://randomuser.me/api/portraits/thumb/women/2.jpg',
      },
    ]
  );

  messages.splice(
    0,
    Infinity,
    ...[
      {
        id: '1',
        content: 'You on your way?',
        createdAt: new Date(new Date('1-1-2019').getTime() - 60 * 1000 * 1000),
        sender: '1',
        recipient: '2',
      },
      {
        id: '2',
        content: "Hey, it's me",
        createdAt: new Date(
          new Date('1-1-2019').getTime() - 2 * 60 * 1000 * 1000
        ),
        sender: '1',
        recipient: '3',
      },
      {
        id: '3',
        content: 'I should buy a boat',
        createdAt: new Date(
          new Date('1-1-2019').getTime() - 24 * 60 * 1000 * 1000
        ),
        sender: '1',
        recipient: '4',
      },
      {
        id: '4',
        content: 'This is wicked good ice cream.',
        createdAt: new Date(
          new Date('1-1-2019').getTime() - 14 * 24 * 60 * 1000 * 1000
        ),
        sender: '1',
        recipient: '5',
      },
    ]
  );

  chats.splice(
    0,
    Infinity,
    ...[
      {
        id: '1',
        participants: ['1', '2'],
        messages: ['1'],
      },
      {
        id: '2',
        participants: ['1', '3'],
        messages: ['2'],
      },
      {
        id: '3',
        participants: ['1', '4'],
        messages: ['3'],
      },
      {
        id: '4',
        participants: ['1', '5'],
        messages: ['4'],
      },
    ]
  );
};

resetDb();
