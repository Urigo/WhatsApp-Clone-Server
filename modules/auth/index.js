"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const core_1 = require("@graphql-modules/core");
const sonar_1 = require("@graphql-modules/sonar");
const typeorm_1 = require("typeorm");
const auth_provider_1 = require("./providers/auth.provider");
const app_symbols_1 = require("../app.symbols");
const apollo_server_express_1 = require("apollo-server-express");
const passport_1 = __importDefault(require("passport"));
const passport_http_1 = __importDefault(require("passport-http"));
const di_1 = require("@graphql-modules/di");
exports.AuthModule = new core_1.GraphQLModule({
    name: "Auth",
    providers: ({ config: { connection, app } }) => [
        { provide: typeorm_1.Connection, useValue: connection },
        { provide: app_symbols_1.APP, useValue: app },
        apollo_server_express_1.PubSub,
        auth_provider_1.AuthProvider,
    ],
    typeDefs: sonar_1.loadSchemaFiles(__dirname + '/schema/'),
    resolvers: sonar_1.loadResolversFiles(__dirname + '/resolvers/'),
    configRequired: true,
    middleware: di_1.InjectFunction(auth_provider_1.AuthProvider, app_symbols_1.APP)((authProvider, app) => {
        passport_1.default.use('basic-signin', new passport_http_1.default.BasicStrategy(async (username, password, done) => {
            done(null, await authProvider.signIn(username, password));
        }));
        passport_1.default.use('basic-signup', new passport_http_1.default.BasicStrategy({ passReqToCallback: true }, async (req, username, password, done) => {
            const name = req.body.name;
            return done(null, !!name && await authProvider.signUp(username, password, name));
        }));
        app.post('/signup', passport_1.default.authenticate('basic-signup', { session: false }), (req, res) => res.json(req.user));
        app.use(passport_1.default.authenticate('basic-signin', { session: false }));
        app.post('/signin', (req, res) => res.json(req.user));
        return {};
    }),
});
