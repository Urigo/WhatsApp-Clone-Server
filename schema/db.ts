import * as moment from "moment";

export enum MessageType {
  TEXT,
  LOCATION,
  PICTURE,
}

export interface User {
  id: number,
  name: string,
  picture?: string | null,
  phone?: string | null,
}

export interface Chat {
  id: number,
  name?: string | null,
  picture?: string | null,
  userIds: number[],
  listingIds: number[],
  memberIds?: number[] | null,
  adminIds?: number[] | null,
  ownerId?: number | null,
  messages: Message[],
}

export interface Message {
  id: number,
  senderId: number,
  content: string,
  createdAt: number,
  type: MessageType,
  recipients: Recipient[],
  holderIds: number[],
}

export interface Recipient {
  id: number,
  receivedAt: number | null,
  readAt: number | null,
}

const users: User[] = [
  {
    id: 0,
    name: 'Ethan Gonzalez',
    picture: 'https://randomuser.me/api/portraits/thumb/men/1.jpg',
    phone: '+391234567890',
  },
  {
    id: 1,
    name: 'Bryan Wallace',
    picture: 'https://randomuser.me/api/portraits/thumb/men/2.jpg',
    phone: '+391234567891',
  },
  {
    id: 2,
    name: 'Avery Stewart',
    picture: 'https://randomuser.me/api/portraits/thumb/women/1.jpg',
    phone: '+391234567892',
  },
  {
    id: 3,
    name: 'Katie Peterson',
    picture: 'https://randomuser.me/api/portraits/thumb/women/2.jpg',
    phone: '+391234567893',
  },
  {
    id: 4,
    name: 'Ray Edwards',
    picture: 'https://randomuser.me/api/portraits/thumb/men/3.jpg',
    phone: '+391234567894',
  },
  {
    id: 5,
    name: 'Niccol√≤ Belli',
    picture: 'https://randomuser.me/api/portraits/thumb/men/4.jpg',
    phone: '+391234567895',
  },
  {
    id: 6,
    name: 'Mario Rossi',
    picture: 'https://randomuser.me/api/portraits/thumb/men/5.jpg',
    phone: '+391234567896',
  },
];

const chats: Chat[] = [
  {
    id: 0,
    name: null,
    picture: null,
    userIds: [0, 2],
    listingIds: [0, 2],
    adminIds: null,
    ownerId: null,
    messages: [
      {
        id: 0,
        senderId: 0,
        content: 'You on your way?',
        createdAt: moment().subtract(1, 'hours').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 2,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [0, 2],
      },
      {
        id: 1,
        senderId: 2,
        content: 'Yep!',
        createdAt: moment().subtract(1, 'hours').add(5, 'minutes').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 0,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [2, 0],
      },
    ],
  },
  {
    id: 1,
    name: null,
    picture: null,
    userIds: [0, 3],
    listingIds: [0, 3],
    adminIds: null,
    ownerId: null,
    messages: [
      {
        id: 0,
        senderId: 0,
        content: 'Hey, it\'s me',
        createdAt: moment().subtract(2, 'hours').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 3,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [0, 3],
      },
    ],
  },
  {
    id: 2,
    name: null,
    picture: null,
    userIds: [0, 4],
    listingIds: [0, 4],
    adminIds: null,
    ownerId: null,
    messages: [
      {
        id: 0,
        senderId: 0,
        content: 'I should buy a boat',
        createdAt: moment().subtract(1, 'days').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 4,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [0, 4],
      },
      {
        id: 1,
        senderId: 0,
        content: 'You still there?',
        createdAt: moment().subtract(1, 'days').add(16, 'hours').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 4,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [0, 4],
      },
    ],
  },
  {
    id: 3,
    name: null,
    picture: null,
    userIds: [2, 3],
    listingIds: [2, 3],
    adminIds: null,
    ownerId: null,
    messages: [
      {
        id: 0,
        senderId: 2,
        content: 'Look at my mukluks!',
        createdAt: moment().subtract(4, 'days').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 3,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [2, 3],
      },
    ],
  },
  {
    id: 4,
    name: null,
    picture: null,
    userIds: [1, 4],
    listingIds: [1, 4],
    adminIds: null,
    ownerId: null,
    messages: [
      {
        id: 0,
        senderId: 1,
        content: 'This is wicked good ice cream.',
        createdAt: moment().subtract(2, 'weeks').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 4,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [1, 4],
      },
      {
        id: 1,
        senderId: 4,
        content: 'Love it!',
        createdAt: moment().subtract(2, 'weeks').add(10, 'minutes').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 1,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [4, 1],
      },
    ],
  },
  {
    id: 5,
    name: null,
    picture: null,
    userIds: [0, 5],
    listingIds: [0],
    adminIds: null,
    ownerId: null,
    messages: [],
  },
  {
    id: 6,
    name: null,
    picture: null,
    userIds: [1, 0],
    listingIds: [1],
    adminIds: null,
    ownerId: null,
    messages: [],
  },
  {
    id: 7,
    name: 'A user 0 group',
    picture: 'https://randomuser.me/api/portraits/thumb/lego/1.jpg',
    userIds: [0, 2, 3, 5],
    listingIds: [0, 2, 3, 5],
    memberIds: [0, 3, 5],
    adminIds: [0, 5],
    ownerId: 0,
    messages: [
      {
        id: 0,
        senderId: 0,
        content: 'I made a group',
        createdAt: moment().subtract(2, 'weeks').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 2,
            receivedAt: null,
            readAt: null,
          },
          {
            id: 3,
            receivedAt: moment().subtract(2, 'weeks').add(1, 'minutes').unix(),
            readAt: moment().subtract(2, 'weeks').add(5, 'minutes').unix(),
          },
          {
            id: 5,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [0, 2, 3, 5],
      },
      {
        id: 1,
        senderId: 0,
        content: 'Ops, user 2 was not supposed to be here',
        createdAt: moment().subtract(2, 'weeks').add(2, 'minutes').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 3,
            receivedAt: moment().subtract(2, 'weeks').add(3, 'minutes').unix(),
            readAt: moment().subtract(2, 'weeks').add(5, 'minutes').unix(),
          },
          {
            id: 5,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [0, 3, 5],
      },
      {
        id: 2,
        senderId: 3,
        content: 'Awesome!',
        createdAt: moment().subtract(2, 'weeks').add(10, 'minutes').unix(),
        type: MessageType.TEXT,
        recipients: [
          {
            id: 0,
            receivedAt: null,
            readAt: null,
          },
          {
            id: 5,
            receivedAt: null,
            readAt: null,
          },
        ],
        holderIds: [0, 3, 5],
      },
    ],
  },
  {
    id: 8,
    name: 'A user 5 group',
    picture: null,
    userIds: [5, 2],
    listingIds: [5, 2],
    memberIds: [5, 2],
    adminIds: [5],
    ownerId: 5,
    messages: [],
  },
];

export const db = {users, chats};
