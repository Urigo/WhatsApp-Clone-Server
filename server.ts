import 'reflect-metadata';
import { ApolloServer } from 'apollo-server-express';
import { GraphQLModule } from '@graphql-modules/core';
import cookie from 'cookie';

import usersModule from './modules/users';
import chatsModule from './modules/chats';

export const rootModule = new GraphQLModule({
  name: 'root',
  imports: [usersModule, chatsModule],
});

const server = new ApolloServer({
  schema: rootModule.schema,
  context: (session: any) => {
    if (session.connection) {
      const req = session.connection.context.session.request;
      const cookies = req.headers.cookie;

      if (cookies) {
        req.cookies = cookie.parse(cookies);
      }
    }

    return rootModule.context(session);
  },
  subscriptions: rootModule.subscriptions,
});
