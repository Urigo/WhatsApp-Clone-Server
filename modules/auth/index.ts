import { GraphQLModule } from '@graphql-modules/core'
import { loadResolversFiles, loadSchemaFiles } from '@graphql-modules/sonar'
import { Express } from 'express'
import { Connection } from 'typeorm'
import { AuthProvider } from './providers/auth.provider'
import { APP } from '../app.symbols'
import { PubSub } from 'apollo-server-express'
import passport from 'passport'
import basicStrategy from 'passport-http'
import { InjectFunction } from '@graphql-modules/di'

export interface IAppModuleConfig {
  connection: Connection
  app: Express
}

export const AuthModule = new GraphQLModule<IAppModuleConfig>({
  name: 'Auth',
  providers: ({ config: { connection, app } }) => [
    { provide: Connection, useValue: connection },
    { provide: APP, useValue: app },
    PubSub,
    AuthProvider,
  ],
  typeDefs: loadSchemaFiles(__dirname + '/schema/'),
  resolvers: loadResolversFiles(__dirname + '/resolvers/'),
  configRequired: true,
  middleware: InjectFunction(AuthProvider, APP)((authProvider, app: Express) => {
    passport.use(
      'basic-signin',
      new basicStrategy.BasicStrategy(async (username: string, password: string, done: any) => {
        done(null, await authProvider.signIn(username, password))
      })
    )

    passport.use(
      'basic-signup',
      new basicStrategy.BasicStrategy(
        { passReqToCallback: true },
        async (
          req: Express.Request & { body: { name?: string } },
          username: string,
          password: string,
          done: any
        ) => {
          const name = req.body.name
          return done(null, !!name && (await authProvider.signUp(username, password, name)))
        }
      )
    )

    app.post('/signup', passport.authenticate('basic-signup', { session: false }), (req, res) =>
      res.json(req.user)
    )

    app.use(passport.authenticate('basic-signin', { session: false }))

    app.post('/signin', (req, res) => res.json(req.user))
    return {}
  }),
})
