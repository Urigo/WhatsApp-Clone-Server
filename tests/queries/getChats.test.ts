import { createTestClient } from 'apollo-server-testing';
import { gql } from 'apollo-server-express';
import { server } from '../../server';
import { resetDb } from '../../db';
import { mockAuth } from '../mocks/auth.provider';

describe('Query.chats', () => {
  beforeEach(resetDb);

  it('should fetch all chats', async () => {
    mockAuth(1);

    const { query } = createTestClient(server);

    const res = await query({
      query: gql`
        query GetChats {
          chats {
            id
            name
            picture
            lastMessage {
              id
              content
              createdAt
            }
          }
        }
      `,
    });

    expect(res.data).toBeDefined();
    expect(res.errors).toBeUndefined();
    expect(res.data).toMatchSnapshot();
  });
});
