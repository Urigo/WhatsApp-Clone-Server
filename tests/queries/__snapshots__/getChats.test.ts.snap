// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Query.chats should fetch all chats 1`] = `
Object {
  "chats": Array [
    Object {
      "id": "1",
      "lastMessage": Object {
        "content": "You on your way?",
        "createdAt": "2018-12-31T05:20:00.000Z",
        "id": "1",
      },
      "name": "Ethan Gonzalez",
      "picture": "https://randomuser.me/api/portraits/thumb/men/1.jpg",
    },
    Object {
      "id": "2",
      "lastMessage": Object {
        "content": "Hey, it's me",
        "createdAt": "2018-12-30T12:40:00.000Z",
        "id": "2",
      },
      "name": "Bryan Wallace",
      "picture": "https://randomuser.me/api/portraits/thumb/men/2.jpg",
    },
    Object {
      "id": "3",
      "lastMessage": Object {
        "content": "I should buy a boat",
        "createdAt": "2018-12-15T06:00:00.000Z",
        "id": "3",
      },
      "name": "Avery Stewart",
      "picture": "https://randomuser.me/api/portraits/thumb/women/1.jpg",
    },
    Object {
      "id": "4",
      "lastMessage": Object {
        "content": "This is wicked good ice cream.",
        "createdAt": "2018-05-12T14:00:00.000Z",
        "id": "4",
      },
      "name": "Katie Peterson",
      "picture": "https://randomuser.me/api/portraits/thumb/women/2.jpg",
    },
  ],
}
`;
